File: TransCoder_tokenized_test_set_functions/java/ARRAY_RANGE_QUERIES_ELEMENTS_FREQUENCY_VALUE.java
static int solveQuery ( int start, int end, int arr [ ] ) {
  Map < Integer, Integer > mp = new HashMap < > ( );
  for ( int i = start;
  i <= end;
  i ++ ) mp . put ( arr [ i ], mp . get ( arr [ i ] ) == null ? 1 : mp . get ( arr [ i ] ) + 1 );
  int count = 0;
  for ( Map . Entry < Integer, Integer > entry : mp . entrySet ( ) ) if ( entry . getKey ( ) == entry . getValue ( ) ) count ++;
  return count;
}

File: results/py/ARRAY_RANGE_QUERIES_ELEMENTS_FREQUENCY_VALUE.cpp
int f_filled ( int start , int end , int arr [ ] ) {
  map < int , int > frequency ;
  for ( int i = start ;
  i <= end ;
  i ++ ) {
    if ( arr [ i ] == frequency . find ( arr [ i ] ) ) frequency [ arr [ i ] ] ++ ;
    else frequency [ arr [ i ] ] = 1 ;
  }
  int count = 0 ;
  for ( int x : frequency ) {
    if ( x == frequency [ x ] ) count ++ ;
  }
  return count ;
}
